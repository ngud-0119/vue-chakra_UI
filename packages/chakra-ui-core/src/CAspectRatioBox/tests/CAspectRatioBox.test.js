import { CAspectRatioBox, CBox } from '../..'
import { render, screen, getElementStyles } from '@/tests/test-utils'
const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { CAspectRatioBox, CBox },
    template: `
    <CAspectRatioBox maxW="400px" data-testid="aspectRatioBox" ${inlineAttrs}>
      <CBox data-testid="child">
        <CBox as="img" src="https://bit.ly/naruto-sage" alt="naruto" objectFit="cover" h="100%" w="100%" data-testid="image" />
      </CBox>
    </CAspectRatioBox>
    `,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const inlineAttrs = ':ratio="1"'
  const { asFragment } = renderComponent({ inlineAttrs })
  expect(asFragment()).toMatchSnapshot()

  const [emotionClassName] = [...screen.getByTestId('aspectRatioBox').classList]
  const pseudoStyles = getElementStyles(`.${emotionClassName}::before`)

  expect(pseudoStyles).toContain(
    `
    padding-bottom: 100%
  `.trim()
  )
})

it('should have correct styles', () => {
  const inlineAttrs = ':ratio="2"'
  renderComponent({ inlineAttrs })

  const [emotionClassName] = [...screen.getByTestId('aspectRatioBox').classList]
  const pseudoStyles = getElementStyles(`.${emotionClassName}::before`)

  expect(pseudoStyles).toContain(
    `
    padding-bottom: 50%
  `.trim()
  )
})
