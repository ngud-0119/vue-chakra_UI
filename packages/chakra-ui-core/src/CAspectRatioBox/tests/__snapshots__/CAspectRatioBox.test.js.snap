// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  max-width: 400px;
}

.emotion-0::before {
  height: 0px;
  content: "";
  display: block;
  padding-bottom: 100%;
}

.emotion-1 {
  position: absolute;
  width: var(--chakra-sizes-full);
  height: var(--chakra-sizes-full);
  top: var(--chakra-space-0);
  left: var(--chakra-space-0);
}

.emotion-2 {
  object-fit: cover;
  height: 100%;
  width: 100%;
}

<div
    class="emotion-0"
    data-chakra-component="CAspectRatioBox"
    data-testid="aspectRatioBox"
  >
    <div
      class="emotion-1"
      data-chakra-component="CBox"
      data-testid="child"
    >
      <img
        alt="naruto"
        class="emotion-2"
        data-chakra-component="CBox"
        data-testid="image"
        src="https://bit.ly/naruto-sage"
      />
    </div>
  </div>
</DocumentFragment>
`;
